utils/alphabet.ts

```diff
@@ -0,0 +1,446 @@
+import { compare as createNaturalCompare } from 'natural-orderby'
+
+interface Character {
+  uppercaseCharacterCodePoint?: number
+  lowercaseCharacterCodePoint?: number
+  codePoint: number
+  value: string
+}
+
+type UnicodePlaneName =
+  | 'supplementaryMultilingual'
+  | 'supplementaryIdeographic'
+  | 'tertiaryIdeographic'
+  | 'basicMultilingual'
+
+interface UnicodePlane {
+  start: number
+  end: number
+}
+
+/**
+ * Utility class to build alphabets
+ */
+export class Alphabet {
+  /**
+   * https://en.wikipedia.org/wiki/Plane_(Unicode)
+   */
+  private static readonly _PLANES: Record<UnicodePlaneName, UnicodePlane> = {
+    supplementaryMultilingual: { start: 0x10000, end: 0x1ffff }, // Supplementary Multilingual Plane (SMP)
+    supplementaryIdeographic: { start: 0x20000, end: 0x2ffff }, // Supplementary Ideographic Plane (SIP)
+    tertiaryIdeographic: { start: 0x30000, end: 0x3ffff }, // Tertiary Ideographic Plane
+    basicMultilingual: { start: 0x0000, end: 0xffff }, // Basic Multilingual Plane (BMP)
+  }
+
+  private _characters: Character[] = []
+
+  private constructor(characters: Character[]) {
+    this._characters = characters
+  }
+
+  /**
+   * Generates an alphabet from the given characters
+   * @param {string|string[]} values - The characters to generate the alphabet from
+   * @returns {Alphabet} - The wrapped alphabet
+   */
+  public static generateFrom(values: string[] | string): Alphabet {
+    let arrayValues = typeof values === 'string' ? [...values] : values
+    if (arrayValues.length !== new Set(arrayValues).size) {
+      throw new Error('The alphabet must not contain repeated characters')
+    }
+    if (arrayValues.some(value => value.length !== 1)) {
+      throw new Error('The alphabet must contain single characters')
+    }
+    return new Alphabet(
+      arrayValues.map(value =>
+        Alphabet._convertCodepointToCharacter(value.codePointAt(0)!),
+      ),
+    )
+  }
+
+  /**
+   * Generates an alphabet containing relevant characters from the Unicode standard
```

Comment:
It's not critical, but I'd like us to try to stay within the 80 character limit for string length.

This is necessary for the convenience of working with the code on laptops with a small screen.

I use the following setting in VS Code:

```json
{
  "editor.rulers": [80]
}
```

Reply:
@azat-io My bad! I fixed it in 39b649f859e9b79acc7bac7ebe702e70224e7b80. I try to remember it but `prettier` doesn't auto-fix comments unfortunately.